app/models/otherReliefs/CreateOrUpdateOtherReliefsModel.scala

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2025 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.otherReliefs
+
+import play.api.libs.json.{Json, OWrites, Reads}
+
+case class CreateOrUpdateOtherReliefsModel(nonDeductableLoanInterest: Option[NonDeductableLoanInterestModel],
+                                           payrollGiving: Option[PayrollGivingModel],
+                                           qualifyingDistributionRedemptionOfSharesAndSecurities: Option[QualifyingDistributionModel],
+                                           maintenancePayments: Option[Seq[MaintenancePaymentsModel]],
+                                           postCessationTradeReliefAndCertainOtherLosses: Option[Seq[PostCessationTradeReliefModel]],
+                                           annualPaymentsMade: Option[AnnualPaymentsMadeModel],
+                                           qualifyingLoanInterestPayments: Option[Seq[QualifyingLoanInterestPaymentsModel]])
+
+object CreateOrUpdateOtherReliefsModel {
+  implicit val reads: Reads[CreateOrUpdateOtherReliefsModel] = Json.reads[CreateOrUpdateOtherReliefsModel]
+
+  implicit val writes: OWrites[CreateOrUpdateOtherReliefsModel] = OWrites[CreateOrUpdateOtherReliefsModel] { model =>
+    val removedEmpty = model.copy(
+      maintenancePayments = if (model.maintenancePayments.exists(_.isEmpty)) None else model.maintenancePayments,
+      postCessationTradeReliefAndCertainOtherLosses = if (model.postCessationTradeReliefAndCertainOtherLosses.exists(_.isEmpty)) None else model.postCessationTradeReliefAndCertainOtherLosses,
+      qualifyingLoanInterestPayments = if (model.qualifyingLoanInterestPayments.exists(_.isEmpty)) None else model.qualifyingLoanInterestPayments
+    )
+
+    Json.writes[CreateOrUpdateOtherReliefsModel].writes(removedEmpty)
```

Comment:
Does this actually work? I'm worried you'll end up with an endless looping stack overflow - because won't these writes use the implicit writes within this model which then means it will call itself recursively?

Reply:
Seems it must work, because the tests cover it. Interesting. I guess, the implicit isn't actually in scope within itself so then it uses the default play writes.

Reply:
It did manage to get it working and added some new tests to the model spec to double check empty seq() were being transformed though I have no idea if it's the best way of handling it